pkg/stream/parse.go

```
@@ -0,0 +1,67 @@
+package stream
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Choice struct {
+	Delta struct {
+		Content string `json:"content"`
+	} `json:"delta"`
+}
+
+type Data struct {
+	Choices []Choice `json:"choices"`
+}
+
+func ParseFile(filename string) error {
+	// Open the file
+	file, err := os.Open(filename)
+	if err != nil {
+		return fmt.Errorf("could not open file: %w", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var contentBuilder strings.Builder
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if the line has "data: " prefix
+		if strings.HasPrefix(line, "data: ") {
+			// Remove the "data: " prefix
+			line = strings.TrimPrefix(line, "data: ")
+		} else {
+			continue // skip lines without "data: "
+		}
+
+		// Parse the JSON line into our `Data` struct
+		var data Data
+		err := json.Unmarshal([]byte(line), &data)
+		if err != nil {
+			// Skip this line if JSON is incomplete or malformed
+			continue
```

I am actually not sure if we want to silently skip this. I'd want to know if the stream is invalid somehow. You will have to deal with two special case:
1. `data: [DONE]`
2. data line being empty

---

Note; I see you made the updates to check for 1 and 2, but we are still silently skipping everything else. If we fail to parse any other data line it should fail and not continue.

---

Ah, I misunderstood. Will make a new pr to address